modules/pytket-stim/pytket/extensions/stim/backends/stim_backend.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2020-2021 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import cast, List, Optional, Sequence, Union
+from uuid import uuid4
+from pytket.backends import (
+    Backend,
+    CircuitNotRunError,
+    CircuitStatus,
+    ResultHandle,
+    StatusEnum,
+)
+from pytket.backends.backendresult import BackendResult
+from pytket.backends.resulthandle import _ResultIdTuple
+from pytket.circuit import Circuit, OpType  # type: ignore
+from pytket.passes import (  # type: ignore
+    BasePass,
+    DecomposeBoxes,
+    FlattenRegisters,
+    RebaseCustom,
+    RemoveRedundancies,
+    SequencePass,
+)
+from pytket.predicates import (  # type: ignore
+    DefaultRegisterPredicate,
+    GateSetPredicate,
+    NoClassicalControlPredicate,
+    Predicate,
+)
+from pytket.utils.outcomearray import OutcomeArray
+from pytket.utils.results import KwargTypes
+import numpy as np
+import stim  # type: ignore
+
+_gate = {
+    OpType.noop: "I",
+    OpType.X: "X",
+    OpType.Y: "Y",
+    OpType.Z: "Z",
+    OpType.H: "H",
+    OpType.S: "S",
+    OpType.SX: "SQRT_X",
+    OpType.SXdg: "SQRT_X_DAG",
+    OpType.CX: "CX",
+    OpType.CY: "CY",
+    OpType.CZ: "CZ",
+    OpType.ISWAPMax: "ISWAP",
+    OpType.SWAP: "SWAP",
+    OpType.Measure: "M",
+    OpType.Reset: "R",
+}
+
+
+def _int_double(x: float) -> int:
+    # return (2x) mod 8 if x is close to a half-integer, otherwise error
+    y = 2 * x
+    n = int(np.round(y))  # type: ignore
+    if np.isclose(y, n):
+        return n % 8
+    else:
+        raise ValueError("Non-Clifford angle encountered")
+
+
+def _tk1_to_cliff(a: float, b: float, c: float) -> Circuit:
+    # Convert Clifford tk1(a, b, c) to a circuit composed of H and S gates
+    n_a, n_b, n_c = _int_double(a), _int_double(b), _int_double(c)
+    circ = Circuit(1)
+    for _ in range(n_c):
+        circ.S(0)
+    for _ in range(n_b):
+        circ.H(0).S(0).H(0)
+    for _ in range(n_a):
+        circ.S(0)
+    circ.add_phase(-0.25 * (n_a + n_b + n_c))
+    return circ
+
+
+def _process_one_circuit(circ: Circuit, n_shots: int) -> BackendResult:
+    qubits = circ.qubits
+    bits = circ.bits
+    c = stim.Circuit()
+    readout_bits = []
+    for cmd in circ.get_commands():
+        optype = cmd.op.type
+        args = cmd.args
+        if optype == OpType.Measure:
```

Comment:
Ah yes I remember having to do something similar to work around mid-circuit measurements for SPAM correction